server/index.js

```diff
@@ -49,15 +42,27 @@ app.get('/products/:rootIsbn/alsoBought', (req, res) => {
       ];
 
       let combinedData = bulkData.map((bulk, index) => {
-        return {
-          _id: bulk.id,
-          isbn: bulk.isbn,
-          title: bulk.title,
-          author: bulk.author,
-          genre: bulk.genre,
-          avgRating: ratingsData[index].data.avgRating,
-          img: randomImages[index]
-        };
+        if (!ratingsData[index]) {
+          return {
+            _id: bulk.id,
+            isbn: bulk.isbn,
+            title: bulk.title,
+            author: bulk.author,
+            genre: bulk.genre,
+            avgRating: 0,
+            img: randomImages[index]
+          };
+        } else {
+          return {
+            _id: bulk.id,
+            isbn: bulk.isbn,
+            title: bulk.title,
+            author: bulk.author,
+            genre: bulk.genre,
+            avgRating: ratingsData[index].data.avgRating,
```

Comment:
Use a short circuit to avoid duplicate code. Or use a library like lodash which has _.get(). Code will look like avgRating: (_.get() || 0) or as part of the _.get set a default of 0